core/common/src/Buffer.kt

```diff
@@ -706,11 +709,18 @@ public class Buffer : Source, Sink {
  */
 @PublishedApi
 @JvmSynthetic
+@OptIn(ExperimentalContracts::class)
 internal inline fun <T> Buffer.seek(
     fromIndex: Long,
     lambda: (Segment?, Long) -> T
 ): T {
-    if (this.head == null) lambda(null, -1L)
+    contract {
+        callsInPlace(lambda, EXACTLY_ONCE)
+    }
+
+    if (this.head == null) {
+        return lambda(null, -1L)
+    }
```

Comment:
This seems like it was a bug. Prior to this, the lambda would be called twice. Once with `(null, -1)` and then again in the `else` with `(null, 0)`. Contract validation failed without changing the implementation to early-return here, as I expect was intended.

Reply:
It was a bug, indeed. Thanks for catching and fixing it!